plugins/languages/sniprun.nix

```diff
@@ -79,61 +166,54 @@ in
-          ctermfg = "Black";
-        };
-        SniprunFloatingWinErr = {
-          fg = "#881515";
-          ctermfg = "DarkRed";
+      helpers.defaultNullOpts.mkNullable' {
+        description = ''
+          Customize highlight groups (setting this overrides colorscheme)
+          any parameters of `nvim_set_hl()` can be passed as-is.
+        '';
+        type = types.submodule {
+          freeformType = types.attrsOf types.anything;
+          options = mapAttrs (optionName: colorOption) {
+            SniprunVirtualTextOk = {
+              bg = "#66eeff";
+              fg = "#000000";
+              ctermbg = "Cyan";
+              ctermfg = "Black";
+            };
+            SniprunFloatingWinOk = {
+              fg = "#66eeff";
+              ctermfg = "Cyan";
+            };
+            SniprunVirtualTextErr = {
+              bg = "#881515";
+              fg = "#000000";
+              ctermbg = "DarkRed";
+              ctermfg = "Black";
+            };
+            SniprunFloatingWinErr = {
+              fg = "#881515";
+              ctermfg = "DarkRed";
+            };
+          };
         };
       };
 
-    liveModeToggle = helpers.defaultNullOpts.mkStr "off" "Live mode toggle, see Usage - Running for more info.";
+    live_mode_toggle = helpers.defaultNullOpts.mkStr "off" ''
+      Live mode toggle, see [Usage - Running] for more info.
 
-    borders = helpers.defaultNullOpts.mkBorder "single" "floating windows" "";
-  };
+      [Usage - Running]: https://michaelb.github.io/sniprun/sources/README.html#running
+    '';
 
-  config = mkIf cfg.enable {
-    extraPlugins =
-      with pkgs.vimPlugins;
-      [ cfg.package ]
-      ++ (optional ((cfg.display != null) && (any (hasPrefix "NvimNotify") cfg.display)) nvim-notify);
+    inline_messages = helpers.defaultNullOpts.mkBool false ''
+      Boolean toggle for a one-line way to display messages
+      to workaround sniprun not being able to display anything.
+    '';
 
-    extraConfigLua =
-      let
-        options = {
-          selected_interpreters = cfg.selectedInterpreters;
-          repl_enable = cfg.replEnable;
-          repl_disable = cfg.replDisable;
-          interpreter_options = cfg.interpreterOptions;
-          inherit (cfg) display;
-          live_display = cfg.liveDisplay;
-          display_options = with cfg.displayOptions; {
-            terminal_width = terminalWidth;
-            notification_timeout = notificationTimeout;
-          };
-          show_no_output = cfg.showNoOutput;
-          inherit (cfg) snipruncolors;
-          live_mode_toggle = cfg.liveModeToggle;
-          inherit (cfg) borders;
-        } // cfg.extraOptions;
-      in
-      ''
-        require('sniprun').setup(${helpers.toLuaObject options})
-      '';
+    borders = helpers.defaultNullOpts.mkEnum [
+      "none"
+      "single"
+      "double"
+      "shadow"
+    ] "single" "Display borders around floating windows.";
   };
 }
```

Comment:
Could you please add a `settingsExample` ?

Reply:
I'm not familiar with the plugin and the docs are extensive, with many _small_ examples. I'll try and cobble something together unless you have a suggestion.

Reply:
> I'm not familiar with the plugin and the docs are extensive, with many _small_ examples. I'll try and cobble something together unless you have a suggestion.

What I often do is to go on `github.com` and look for `language:lua require('foo').setup` and browse the "Code" results.
It doesn't have to be a realistic config.

Reply:
Thanks for the tip. I have a small example based on the docs, but it could of course be improved by looking at [real configs](https://github.com/search?q=language%3Alua+require%28%27sniprun%27%29.setup&type=code).